src/components/pages/home/tab/profile/index.tsx

```diff
@@ -14,41 +14,60 @@ import WhyMe from "./whyMe";
 
 import contractAddresses from '../../../../../config/ines.fund';
 
+const ONETHOUSAND = 1000;
+
+const calcCummulativePayments = (repayment) => {
+  return repayment.map( (payment, index) =>
+    Object.assign({}, payment, {
+      payment: repayment.slice(0, index+1).reduce((a,b) => a+Number(b.total), 0) 
+    })
+  )
+}
 
 class Profile extends React.Component<{}> {
   state = {
     repayments: []
   }
 
-  componentDidMount = async () => {
-    const networkId = await getNetworkId();
-    const termsContractAddress = contractAddresses[networkId]['TermsContract'];
-    const termsContractInstance = await getContractInstance(
-      TermsContract.abi,
-      termsContractAddress
-    );
-    try {
-      const loanParams = await contractMethodCall(termsContractInstance, 'getLoanParams');
-      const numScheduledPayments = parseInt(loanParams.loanPeriod);
+  componentDidMount =  () => {
+    setTimeout( async() => {
+      const termsContractInstance = await getDeployedFromConfig('TermsContract', contractAddresses);
+
+      try {
+        const loanParams = await getLoanParams(termsContractInstance);
+        const paymentToken = await getTokenDetailsFromAddress(loanParams.principalToken);
+        const numScheduledPayments = parseInt(loanParams.loanPeriod);
+        const prepDueTimestamp = (dueTimestamp, startTimestamp) => (dueTimestamp * ONETHOUSAND) + (startTimestamp == 0 ? new Date().getTime() : 0);
+
+        let repayments = await Promise.all(
+          Array(numScheduledPayments)
+          .fill({})
+          .map(async (element, index) => {
+            const requestedScheduledPayment = await getRequestedScheduledPayment(termsContractInstance, index + 1);
+            const scheduledPayment = await getScheduledPayment(termsContractInstance, index + 1);
+            const combined = Object.assign({}, scheduledPayment, requestedScheduledPayment );
+            return {
+              due: prepDueTimestamp(combined.dueTimestamp, loanParams.loanStartTimestamp),
+              interest: prepBigNumber(combined.interestPayment,paymentToken.decimals,true),
+              principal: prepBigNumber(combined.principalPayment,paymentToken.decimals,true),
+              total: prepBigNumber(combined.totalPayment,paymentToken.decimals,true)
+            };
+          })
+        );
+        repayments = calcCummulativePayments(repayments);
 
-      const repayments = await Promise.all(
-        Array(numScheduledPayments)
-        .fill({})
-        .map(async (element, index) => {
-          const {due, interest, principal, total} = await termsContractInstance.methods.getScheduledPayment(index + 1).call();
-          return {due, interest, principal, total};
-        })
-      );
-    this.setState({repayments});
-    } catch (err) {
-      console.log(err)
-    }
+      this.setState({repayments});
+      } catch (err) {
+        console.log(err)
+      }
+    }, 3000);
```

Comment:
Why do you use setTimeOut? @adibas03

Reply:
I noticed Metamask has an issue with collusion of RPC calls, which was causing the Node Errors we were getting intermittently. The timeout stops that temporarily, while we think out a permanent fix